packages/graphiql/src/utility/dset-merger.ts

```diff
@@ -0,0 +1,52 @@
+// @ts-nocheck
+
+/*
+!!IMPORTANT
+This is a _fork_ of https://github.com/lukeed/dset/ that adds support in for deep-merging collisions. Once `dset/merge` becomes a thing we can remove this.
+
+@maraisr will be monitoring this.
+ */
+
+export function dset<T extends object, V>(
+  obj: T,
+  keys: string | ArrayLike<string | number>,
+  value: V,
+): void;
+export function dset(obj, keys, val) {
```

Comment:
@lukeed with your blessing I have in-lined this for now, once `dset/merge` if/when becomes a think I'll update this.